impl/src/main/java/impl/AnswerResourceImpl.java

```diff
@@ -2,25 +2,57 @@
 
 import api.Answer;
 import api.AnswerResource;
+import api.auth.Auth;
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
 import dao.AnswerDao;
+import dao.QuestionDao;
+import io.netty.handler.codec.http.HttpResponseStatus;
 import rx.Observable;
+import se.fortnox.reactivewizard.db.transactions.DaoTransactions;
+import se.fortnox.reactivewizard.jaxrs.WebException;
+
+import java.util.List;
+import java.util.Objects;
+
+import static java.util.Arrays.asList;
 
 @Singleton
 public class AnswerResourceImpl implements AnswerResource {
 
-    private final AnswerDao answerDao;
+    private final AnswerDao       answerDao;
+    private final QuestionDao     questionDao;
+    private final DaoTransactions daoTransactions;
 
     @Inject
-    public AnswerResourceImpl(AnswerDao answerDao) {
-
+    public AnswerResourceImpl(AnswerDao answerDao, QuestionDao questionDao, DaoTransactions daoTransactions) {
         this.answerDao = answerDao;
+        this.questionDao = questionDao;
+        this.daoTransactions = daoTransactions;
     }
 
     @Override
-    public Observable<Answer> getAnswers(long questionId) {
-        return answerDao.getAnswers(questionId);
+    public Observable<Answer> answerQuestion(Auth auth, Answer answer, long questionId) {
+        Objects.requireNonNull(answer.getAnswer());
+
+        //If title is missing we just extract a title from the answer
+        if (answer.getTitle() == null) {
+            answer.setTitle(answer.getAnswer().substring(0, Math.min(10, answer.getAnswer().length())) + "...");
```

Comment:
answer.getAnswer() -> nullpointer maybe?

Reply:
But I check for null in row 36
```
Objects.requireNonNull(answer.getAnswer())
```